### PR TITLE
313 force insert failure

### PR DESCRIPTION
**Pull request by ricellis**
*What*

Closed datastores left open in `DatastoreImplForceInsertTest`.

*Why*

Leaving the datastores open could lead to leaked connections and later thread deaths from `udpateSchema` failures e.g.
>E/AndroidRuntime(21513): android.database.sqlite.SQLiteException: cannot rollback - no transaction is active (code 1)
...
com.cloudant.sync.sqlite.android.AndroidSQLite.endTransaction(AndroidSQLite.java:76)
E/AndroidRuntime(21513): 	at com.cloudant.sync.sqlite.SQLDatabaseFactory.updateSchema(SQLDatabaseFactory.java:206)

*How*

Added `datastore.close()` call to test teardown. This exposed some other tests failing to close datastores correctly, namely:
`DatastoreManagerTest`

and an issue where calling `close` more than once on the same `Datastore` resulted in a `RejectedExecutionException`.

*Test*

This is mainly changes to existing tests to close datatstores opened by the tests.

### CODE REVIEW

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/sqlite/SQLDatabaseQueue.java [hunk 1]
```diff
@@ -143,21 +144,22 @@ public Integer call(SQLDatabase db) throws Exception {
...
      */
     public void shutdown() {
-        acceptTasks = false;
-        //pass straight to queue, tasks passed via submitTaskToQueue will now be blocked.
-        queue.submit(new Runnable() {
-            @Override
-            public void run() {
-                db.close();
+        // If shutdown has already been called then we don't need to shutdown again
+        if (acceptTasks.getAndSet(false)) {
```
Comment (tomblench) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/sqlite/SQLDatabaseQueue.java]: Do you think we should have a warning if shutdown is called more than once? As discussed it's harmless but having a warning might help spot bad patterns in client code.

Comment (ricellis) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/sqlite/SQLDatabaseQueue.java]: Added in 9ae4ab5

**Comment by tomblench** [comment]
+1

**Comment by rhyshort** [comment]
+1

**Pull request merged**